schemas/config/alignment_general.schema.yaml

```diff
@@ -32,25 +32,31 @@ properties:
   # aligner
   aligner:
     description: which aligner to use
-    default: bwa
-    enum: [bowtie2, bwa, hisat2, salmon]
-    type: string
-
-  # bwa alignment parameters
-  bwa_algo:
-    description: the algorithm bwa index uses for alignment
-    default: bwtsw
-    type: string
+    type: object
+    properties:
+      bowtie2:
+        properties:
+          index: {type: string}
+          align: {type: string}
+      bwa:
+        properties:
+          index: {type: string}
+          align: {type: string}
+      hisat2:
+        properties:
+          index: {type: string}
+          align: {type: string}
+      salmon:
+        properties:
+          index: {type: string}
+          align: {type: string}
+    minProperties: 1
+    additionalProperties: false
+    default:
+      bwa:
+        index: -a bwtsw
+        align: ''
 
-  # salmon mapping
-  salmon_index:
-    description: optional arguments for salmon index, such as -k, --gencode and -decoys {decoyfile}
-    default: ''
-    type: string
-  salmon_aln:
-    description: optional arguments for salmon alignment
-    default: --seqBias --gcBias --validateMappings # --fldMean --fldSD (defaults are in use) # --posBias (currently experimental)
```

Comment:
Defaults make the script way more user friendly. Otherwise time is wasted setting flasg, or they must be hardcoded...

Reply:
See changes